src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -292,6 +321,7 @@ FindShardIntervalIndex(Datum searchedValue, DistTableCacheEntry *cacheEntry)
 	bool useBinarySearch = (partitionMethod != DISTRIBUTE_BY_HASH ||
 							!cacheEntry->hasUniformHashDistribution);
 	int shardIndex = INVALID_SHARD_INDEX;
+	Oid shardIntervalCollation = cacheEntry->partitionColumn->varcollid;
```

Comment:
Here and various other places: partitionColumn may be NULL in case of reference tables.